### ISSUE TITLE
extract_outlier_frames not working

### ISSUE THREAD

**Issue by JohnCraigPublic** [DeepLabCut/DeepLabCut#339]
Ubuntu 16.04 LTS
DLC 2.063
Conda Env

extract_outlier_frames finds frames to extract, then, when trying to do so, fails with this message:

ValueError: could not broadcast input array from shape (2) into shape (1)

I've successfully used this function on another project, but now having this problem.  This project is a sort of test project in which I am only marking one key-point, in case that matters.

**Comment by MMathisLab** [comment]
Could you provide the full error message please; since it does work for you in the past, it’s likely your video - does it have more than 3 RGB channels?

**Comment by JohnCraigPublic** [comment]
I think the video is a normal 3-channel video in mp4 format.

Here is transcript:
---------------------------------------------------------------------------
```
In [3]: deeplabcut.extract_outlier_frames(config_path,['/home/john/Projects/Epona-John
   ...: Craig-2019-06-23/AnalyzeVideos/3315.mp4'], epsilon=175)
network parameters: DeepCut_resnet50_EponaJun23shuffle1_50000
Method  jump  found  21  putative outlier frames.
Do you want to proceed with extracting  20  of those?In [3]: deeplabcut.extract_outlier_frames(config_path,['/home/john/Projects/Epona-JohnIn [3]: deeplabcut.extract_outlier_frames(config_path,['/home/john/Projects/Epona-John
...: Craig-2019-06-23/AnalyzeVideos/3315.mp4'], epsilon=175)
network parameters: DeepCut_resnet50_EponaJun23shuffle1_50000
Method jump found 21 putative outlier frames.
Do you want to proceed with extracting 20 of those?
If this list is very large, perhaps consider changing the paramters (start, stop, epsilon, comparisonbodyparts) or use a different method.
yes/noyes
Frames from video 3315 already extracted (more will be added)!
Loading video...
Duration of video [s]: 9.766666666666667 , recorded @ 30.0 fps!
Overall # of frames: 293 with (cropped) frame dimensions:
Kmeans-quantization based extracting of frames from 0.0 seconds to 9.77 seconds.
Extracting and downsampling... 21 frames from the video.
21it [00:02, 9.05it/s]
Kmeans clustering ... (this might take a while)
Let's select frames indices: [92, 188, 112, 31, 129, 58, 167, 50, 195, 190, 96, 115, 52, 110, 95, 168, 113, 187, 59, 128]
...: Craig-2019-06-23/AnalyzeVideos/3315.mp4'], epsilon=175)
network parameters: DeepCut_resnet50_EponaJun23shuffle1_50000
Method jump found 21 putative outlier frames.
Do you want to proceed with extracting 20 of those?
If this list is very large, perhaps consider changing the paramters (start, stop, epsilon, comparisonbodyparts) or use a different method.
yes/noyes
Frames from video 3315 already extracted (more will be added)!
Loading video...
Duration of video [s]: 9.766666666666667 , recorded @ 30.0 fps!
Overall # of frames: 293 with (cropped) frame dimensions:
Kmeans-quantization based extracting of frames from 0.0 seconds to 9.77 seconds.
Extracting and downsampling... 21 frames from the video.
21it [00:02, 9.05it/s]
Kmeans clustering ... (this might take a while)
Let's select frames indices: [92, 188, 112, 31, 129, 58, 167, 50, 195, 190, 96, 115, 52, 110, 95, 168, 113, 187, 59, 128]
If this list is very large, perhaps consider changing the paramters (start, stop, epsilon, comparisonbodyparts) or use a different method.
yes/noyes
Frames from video 3315  already extracted (more will be added)!
Loading video...
Duration of video [s]:  9.766666666666667 , recorded @  30.0 fps!
Overall # of frames:  293 with (cropped) frame dimensions: 
Kmeans-quantization based extracting of frames from 0.0  seconds to 9.77  seconds.
Extracting and downsampling... 21  frames from the video.
21it [00:02,  9.05it/s]
Kmeans clustering ... (this might take a while)
Let's select frames indices: [92, 188, 112, 31, 129, 58, 167, 50, 195, 190, 96, 115, 52, 110, 95, 168, 113, 187, 59, 128]

ValueError                                Traceback (most recent call last)
<ipython-input-3-883474474077> in <module>()
----> 1 deeplabcut.extract_outlier_frames(config_path,['/home/john/Projects/Epona-JohnCraig-2019-06-23/AnalyzeVideos/3315.mp4'], epsilon=175)

/home/john/anaconda3/envs/DLC/lib/python3.6/site-packages/deeplabcut/refine_training_dataset/outlier_frames.py in extract_outlier_frames(config, videos, videotype, shuffle, trainingsetindex, outlieralgorithm, comparisonbodyparts, epsilon, p_bound, ARdegree, MAdegree, alpha, extractionalgorithm, automatic, cluster_resizewidth, cluster_color, opencv, savelabeled, destfolder)
    187               if askuser=='y' or askuser=='yes' or askuser=='Ja' or askuser=='ha': # multilanguage support :)
    188                   #Now extract from those Indices!
--> 189                   ExtractFramesbasedonPreselection(Indices,extractionalgorithm,Dataframe,dataname,scorer,video,cfg,config,opencv,cluster_resizewidth,cluster_color,savelabeled)
    190               else:
    191                   print("Nothing extracted, change parameters and start again...")

/home/john/anaconda3/envs/DLC/lib/python3.6/site-packages/deeplabcut/refine_training_dataset/outlier_frames.py in ExtractFramesbasedonPreselection(Index, extractionalgorithm, Dataframe, dataname, scorer, video, cfg, config, opencv, cluster_resizewidth, cluster_color, savelabeled)
    451     for index in frames2pick: ##tqdm(range(0,nframes,10)):
    452         if opencv:
--> 453             PlottingSingleFramecv2(cap,cv2,cfg['cropping'],coords,Dataframe,bodyparts,tmpfolder,index,scorer,cfg['dotsize'],cfg['pcutoff'],cfg['alphavalue'],colors,strwidth,savelabeled)
    454         else:
    455             PlottingSingleFrame(clip,Dataframe,bodyparts,tmpfolder,index,scorer,cfg['dotsize'],cfg['pcutoff'],cfg['alphavalue'],colors,strwidth,savelabeled)

/home/john/anaconda3/envs/DLC/lib/python3.6/site-packages/deeplabcut/refine_training_dataset/outlier_frames.py in PlottingSingleFramecv2(cap, cv2, crop, coords, Dataframe, bodyparts2plot, tmpfolder, index, scorer, dotsize, pcutoff, alphavalue, colors, strwidth, savelabeled)
    568                         Dataframe[scorer][bp]['x'].values[index],
    569                         Dataframe[scorer][bp]['y'].values[index],'.',
--> 570                         color=colors(bpindex),
    571                         ms=dotsize,
    572                         alpha=alphavalue)

/home/john/anaconda3/envs/DLC/lib/python3.6/site-packages/matplotlib/colors.py in __call__(self, X, alpha, bytes)
    455         # See class docstring for arg/kwarg documentation.
    456         if not self._isinit:
--> 457             self._init()
    458         mask_bad = None
    459         if not np.iterable(X):

/home/john/anaconda3/envs/DLC/lib/python3.6/site-packages/matplotlib/colors.py in _init(self)
    661         self._lut = np.ones((self.N + 3, 4), float)
    662         self._lut[:-3, 0] = makeMappingArray(
--> 663             self.N, self._segmentdata['red'], self._gamma)
    664         self._lut[:-3, 1] = makeMappingArray(
    665             self.N, self._segmentdata['green'], self._gamma)

ValueError: could not broadcast input array from shape (2) into shape (1)

In [4]: 
```

**Comment by MMathisLab** [comment]
I think because you are only finding 21 outliers, and you extracted 20 already beforehand, it can't save over the image again. If you want to find more frames to label, perhaps adjust the epsilon to be lower, i.e. 75 and see if you find more than 21 outliers...

**Comment by JohnCraigPublic** [comment]
Hmmm... how do you see that I "extracted 20 already beforehand" ?   If I did, it was accidental and I never dealt with them - so how can I 'start over' ?

**Comment by MMathisLab** [comment]
you can delete the frames from the folder (with video name 3315), and try again... 

I am looking at this output here, which repeats:

```
deeplabcut.extract_outlier_frames(config_path,['/home/john/Projects/Epona-John
   ...: Craig-2019-06-23/AnalyzeVideos/3315.mp4'], epsilon=175)
network parameters: DeepCut_resnet50_EponaJun23shuffle1_50000
Method  jump  found  21  putative outlier frames.
Do you want to proceed with extracting  20  of those?In [3]: deeplabcut.extract_outlier_frames(config_path,['/home/john/Projects/Epona-JohnIn [3]: deeplabcut.extract_outlier_frames(config_path,['/home/john/Projects/Epona-John
...: Craig-2019-06-23/AnalyzeVideos/3315.mp4'], epsilon=175)
network parameters: DeepCut_resnet50_EponaJun23shuffle1_50000
Method jump found 21 putative outlier frames.
Do you want to proceed with extracting 20 of those?
If this list is very large, perhaps consider changing the paramters (start, stop, epsilon, comparisonbodyparts) or use a different method.
yes/noyes
Frames from video 3315 already extracted (more will be added)!
Loading video...
Duration of video [s]: 9.766666666666667 , recorded @ 30.0 fps!
Overall # of frames: 293 with (cropped) frame dimensions:
Kmeans-quantization based extracting of frames from 0.0 seconds to 9.77 seconds.
Extracting and downsampling... 21 frames from the video.
21it [00:02, 9.05it/s]
Kmeans clustering ... (this might take a while)
Let's select frames indices: [92, 188, 112, 31, 129, 58, 167, 50, 195, 190, 96, 115, 52, 110, 95, 168, 113, 187, 59, 128]
```

**Comment by MMathisLab** [comment]
It also seems you are getting a really slim extracted frames, i.e. only 21, so perhaps then pick only 5 of those, vs 20... cutting it close ;)

**Comment by JohnCraigPublic** [comment]
I looked in the folder where it should be writing out the frames - there were some there, so I deleted them and started fresh.  It would write out 2 or 3 or 4, and then crash as before - then I'd run the same command again, and it would write a couple more before crashing.  By running the command about 15 times, I got all the frames written out -- and could go on to merge_dataset and launch training....

So, something a bit spooky going on.  Not sure what it means.  I'm not sweating it for now.  I'll let you know if I see it again and have more data...
